src/main/docs/man/man3/DragonLimelight.3
.TH "DragonLimelight" 3 "Thu Oct 31 2019" "2020 Template Project" \" -*- nroff -*-
.ad l
.nh
.SH NAME
DragonLimelight
.SH SYNOPSIS
.br
.PP
.PP
\fC#include <DragonLimelight\&.h>\fP
.SS "Public Types"

.in +1c
.ti -1c
.RI "enum \fBLED_MODE\fP { \fBLED_DEFUALT\fP, \fBLED_OFF\fP, \fBLED_BLINK\fP, \fBLED_ON\fP }"
.br
.ti -1c
.RI "enum \fBCAM_MODE\fP { \fBCAM_VISION\fP, \fBCAM_DRIVER\fP }"
.br
.ti -1c
.RI "enum \fBSTREAM_MODE\fP { \fBSTREAM_DEFAULT\fP, \fBSTREAM_MAIN_AND_SECOND\fP, \fBSTREAM_SECOND_AND_MAIN\fP }"
.br
.ti -1c
.RI "enum \fBSNAPSHOT_MODE\fP { \fBSNAP_OFF\fP, \fBSNAP_ON\fP }"
.br
.in -1c
.SS "Public Member Functions"

.in +1c
.ti -1c
.RI "double \fBHasTarget\fP ()"
.br
.ti -1c
.RI "double \fBGetTargetHorizontalOffset\fP ()"
.br
.ti -1c
.RI "double \fBGetTargetVerticalOffset\fP ()"
.br
.ti -1c
.RI "double \fBGetTargetArea\fP ()"
.br
.ti -1c
.RI "double \fBGetTargetSkew\fP ()"
.br
.ti -1c
.RI "double \fBGetPipelineLatency\fP ()"
.br
.ti -1c
.RI "std::vector< double > \fBGet3DSolve\fP ()"
.br
.ti -1c
.RI "void \fBSetLEDMode\fP (\fBDragonLimelight::LED_MODE\fP mode)"
.br
.ti -1c
.RI "void \fBSetCamMode\fP (\fBDragonLimelight::CAM_MODE\fP mode)"
.br
.ti -1c
.RI "void \fBSetPipeline\fP (int pipeline)"
.br
.ti -1c
.RI "void \fBSetStreamMode\fP (\fBDragonLimelight::STREAM_MODE\fP mode)"
.br
.ti -1c
.RI "void \fBToggleSnapshot\fP (\fBDragonLimelight::SNAPSHOT_MODE\fP toggle)"
.br
.ti -1c
.RI "void \fBPrintValues\fP ()"
.br
.in -1c
.SS "Static Public Member Functions"

.in +1c
.ti -1c
.RI "static \fBDragonLimelight\fP * \fBGetInstance\fP ()"
.br
.in -1c
.SH "Detailed Description"
.PP 
Definition at line 28 of file DragonLimelight\&.h\&.
.SH "Member Enumeration Documentation"
.PP 
.SS "enum \fBDragonLimelight::CAM_MODE\fP"

.PP
\fBEnumerator\fP
.in +1c
.TP
\fB\fICAM_VISION \fP\fP
.TP
\fB\fICAM_DRIVER \fP\fP
.PP
Definition at line 39 of file DragonLimelight\&.h\&.
.SS "enum \fBDragonLimelight::LED_MODE\fP"

.PP
\fBEnumerator\fP
.in +1c
.TP
\fB\fILED_DEFUALT \fP\fP
.TP
\fB\fILED_OFF \fP\fP
.TP
\fB\fILED_BLINK \fP\fP
.TP
\fB\fILED_ON \fP\fP
.PP
Definition at line 31 of file DragonLimelight\&.h\&.
.SS "enum \fBDragonLimelight::SNAPSHOT_MODE\fP"

.PP
\fBEnumerator\fP
.in +1c
.TP
\fB\fISNAP_OFF \fP\fP
.TP
\fB\fISNAP_ON \fP\fP
.PP
Definition at line 52 of file DragonLimelight\&.h\&.
.SS "enum \fBDragonLimelight::STREAM_MODE\fP"

.PP
\fBEnumerator\fP
.in +1c
.TP
\fB\fISTREAM_DEFAULT \fP\fP
.TP
\fB\fISTREAM_MAIN_AND_SECOND \fP\fP
.TP
\fB\fISTREAM_SECOND_AND_MAIN \fP\fP
.PP
Definition at line 45 of file DragonLimelight\&.h\&.
.SH "Member Function Documentation"
.PP 
.SS "std::vector< double > DragonLimelight::Get3DSolve ()"

.PP
Definition at line 30 of file DragonLimelight\&.cpp\&.
.SS "\fBDragonLimelight\fP * DragonLimelight::GetInstance ()\fC [static]\fP"

.PP
Definition at line 107 of file DragonLimelight\&.cpp\&.
.SS "double DragonLimelight::GetPipelineLatency ()"

.PP
Definition at line 65 of file DragonLimelight\&.cpp\&.
.SS "double DragonLimelight::GetTargetArea ()"

.PP
Definition at line 53 of file DragonLimelight\&.cpp\&.
.SS "double DragonLimelight::GetTargetHorizontalOffset ()"

.PP
Definition at line 41 of file DragonLimelight\&.cpp\&.
.SS "double DragonLimelight::GetTargetSkew ()"

.PP
Definition at line 59 of file DragonLimelight\&.cpp\&.
.SS "double DragonLimelight::GetTargetVerticalOffset ()"

.PP
Definition at line 47 of file DragonLimelight\&.cpp\&.
.SS "double DragonLimelight::HasTarget ()"

.PP
Definition at line 35 of file DragonLimelight\&.cpp\&.
.SS "void DragonLimelight::PrintValues ()"

.PP
Definition at line 97 of file DragonLimelight\&.cpp\&.
.SS "void DragonLimelight::SetCamMode (\fBDragonLimelight::CAM_MODE\fP mode)"

.PP
Definition at line 76 of file DragonLimelight\&.cpp\&.
.SS "void DragonLimelight::SetLEDMode (\fBDragonLimelight::LED_MODE\fP mode)"

.PP
Definition at line 71 of file DragonLimelight\&.cpp\&.
.SS "void DragonLimelight::SetPipeline (int pipeline)"

.PP
Definition at line 81 of file DragonLimelight\&.cpp\&.
.SS "void DragonLimelight::SetStreamMode (\fBDragonLimelight::STREAM_MODE\fP mode)"

.PP
Definition at line 86 of file DragonLimelight\&.cpp\&.
.SS "void DragonLimelight::ToggleSnapshot (\fBDragonLimelight::SNAPSHOT_MODE\fP toggle)"

.PP
Definition at line 92 of file DragonLimelight\&.cpp\&.

.SH "Author"
.PP 
Generated automatically by Doxygen for 2020 Template Project from the source code\&.


src/main/cpp/hw/DragonLimelight.cpp

///====================================================================================================================================================
/// Copyright 2019 Lake Orion Robotics FIRST Team 302
///
/// Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"),
/// to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense,
/// and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
///
/// The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
///
/// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
/// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
/// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE
/// OR OTHER DEALINGS IN THE SOFTWARE.
///====================================================================================================================================================
/*
/*   Created by Austin Szczesniak
/*   Date: 1-19-2019
*/

#include <hw/DragonLimelight.h>
#include <vector>

DragonLimelight* DragonLimelight::m_instance = nullptr;

DragonLimelight::DragonLimelight() :
    table(NetworkTableInstance::GetDefault().GetTable("limelight"))
{}

std::vector<double> DragonLimelight::Get3DSolve()
{
    return table->GetNumberArray("camtran", 0);
}

double DragonLimelight::HasTarget()
{
    double v = table->GetNumber("tv", 0.0);
    return v;
}

double DragonLimelight::GetTargetHorizontalOffset()
{
    double x = table->GetNumber("tx", 0.0);
    return x - 1.25;
}

double DragonLimelight::GetTargetVerticalOffset()
{
    double y = table->GetNumber("ty", 0.0);
    return y;
}

double DragonLimelight::GetTargetArea()
{
    double a = table->GetNumber("ta", 0.0);
    return a;
}

double DragonLimelight::GetTargetSkew()
{
    double s = table->GetNumber("ts", 0.0);
    return s;
}

double DragonLimelight::GetPipelineLatency()
{
    double l = table->GetNumber("tl", 0.0);
    return l;
}

void DragonLimelight::SetLEDMode(DragonLimelight::LED_MODE mode)
{
    table->PutNumber("ledMode", mode);
}

void DragonLimelight::SetCamMode(DragonLimelight::CAM_MODE mode)
{
    table->PutNumber("camMode", mode);
}

void DragonLimelight::SetPipeline(int pipeline)
{
    table->PutNumber("pipeline", pipeline);
}

void DragonLimelight::SetStreamMode(DragonLimelight::STREAM_MODE mode)
{
    table->PutNumber("stream", mode);
}

// MAX of 32 snapshots can be saved
void DragonLimelight::ToggleSnapshot(DragonLimelight::SNAPSHOT_MODE toggle)
{
    table->PutNumber("snapshot", toggle);
}

void DragonLimelight::PrintValues()
{
    printf("Has Target: %f \n", HasTarget());
    printf("X Offset: %f \n", GetTargetHorizontalOffset());
    printf("Y Offset: %f \n", GetTargetVerticalOffset());
    printf("Area: %f \n", GetTargetArea());
    printf("Skew: %f \n", GetTargetSkew());
    printf("Latency: %f \n", GetPipelineLatency());
}

DragonLimelight* DragonLimelight::GetInstance()
{
    if(m_instance == nullptr)
    {
        m_instance = new DragonLimelight();
    }

    return m_instance;
}